lib/url_parser.rb

```diff
@@ -221,4 +221,15 @@ def remove_whitespace
   def remove_quotes
     url.gsub!(/["']/, '')
   end
+
+  def last_field(s, pattern)
+    # "".split(pattern) = [], not [""], but we want to return ""
+    # https://chriszetter.com/blog/2017/10/29/splitting-strings/
+    fields = s.split(pattern)
+    if fields.empty?
+      +""
```

Comment:
```suggestion
      ""
```

Did you mean to leave this `+` here? It's unclear to me what the `+` is acting on with this syntax.

Reply:
it keeps the string from being frozen with `# frozen_string_literal: true` at the top of the file, so we can `gsub!` it later.

Reply:
and yes I had to google how to do that :-)